### ISSUE TITLE
Investigate the effort to migrate from save_step_view to the API v2 upload endpoint

### ISSUE THREAD

**Issue by giohappy** [geosolutions-it/nexus-geonode#550]
The new upload client that @allyoucanmap is prototyping mimics the current [save_step_view](https://github.com/GeoNode/geonode/blob/c8d5e9babbf3d8c852c347c2a62ddde941bfe8fd/geonode/upload/views.py#L133) and its interaction with the [data_upload](https://github.com/GeoNode/geonode/blob/c8d5e9babbf3d8c852c347c2a62ddde941bfe8fd/geonode/upload/views.py#L651) view. 

This flow requires the active and continous POSTing from the web page to move the upload workflow forward, even if no interactive step is required (SRS or time configurations),. This translates in the client having to recurively call the upload view until either an interaction step is requested (with redirection to the corresponding pages) or the upload completes.
At the same time the client polls the API v2 `uploads/` endpoint to track the progress of the upload. 

This is has several drawbacks:

 - wasted bandwidth and resources just for pushing the flow forward
 - doubling the number of requests (pushing the flow and obtaining its progress) on two unrelated endpoints

The `uploads/` API already supports POST requests. It simulates the sequence of requests required by the upload view. Some tests show that it's already able to complete the upload of a file that doesn't require interactions. However, if an interaction step is required the view returned as redirect_url doesn't work as expected. For example, when a shapefile eligable for time dimensions, the time step view returns a JSON response instead of the Time form.

We want to investigate how far we are from having a working API that is able to handle the communication between the client and the upload view, also in case user interactions are needed. To be noticed that we don't want to manage the interactions through the API at this stage. These will still be managed through the legacy views and forms for the time being. 

@italogsfernandes [here you are](https://github.com/geosolutions-it/nexus-geonode/files/7883879/IGAD_Fiber_optic_tersiterialf.zip) an example of a shepfile with time columns that should trigger the Time step. For the CRS step any CSV should work. 
In any case @allyoucanmap can provide additional test datasets.

**Comment by italogsfernandes** [comment]
For files that need intervention, the api is almost working.
The response of following step: [check step](https://github.com/GeoNode/geonode/blob/master/geonode/upload/api/views.py#L105) is ignored by it.
In this response we can check the `required_input` field.
I think we can just:
* Inspect this response to see if `required_input` is not false or null.
* If there we need an intervention, return this response to the user. for example:
```json
{
    "url": "/upload/?id=33",
    "status": "incomplete",
    "required_input": [
        {
            "name": "created_at",
            "binding": "Date"
        },
        {
            "name": "updated_at",
            "binding": "Date"
        }
    ],
    "success": true,
    "id": 33,
    "redirect_to": "http://localhost:8000/upload/time?id=33"
}
```
* Otherwise just follow to the next step as usual.


Another difference, when we have a upload that doesn't need intervention, the api steps are: `None`, `check`, `final`.
The steps pushed by the client are: `None`, `srs`, `check`. But I confirm that they both seems to work in this case.

**Comment by allyoucanmap** [comment]
@italogsfernandes thanks, but currently when I try to use `redirect_to` the page is not rendered but I can see only a json

![image](https://user-images.githubusercontent.com/19175505/150097102-a2de53ab-985e-4282-9cd8-7abe74d4b08c.png)

I would expect to see this template
https://github.com/GeoNode/geonode/blob/master/geonode/upload/views.py#L488

**Comment by giohappy** [comment]
@italogsfernandes apparently when the uploads/ endpoint calls the internal upload_view steps, something inside the upload session differs from what we get when the same steps are triggered from the upload page. 
It looks like the time_step_view doesn't reach [the line where the form should be rendered](https://github.com/GeoNode/geonode/blob/41198fd14c3bfe194fd86d4e55346dd10c8f2106/geonode/upload/views.py#L495)

**Comment by allyoucanmap** [comment]
@giohappy @italogsfernandes I tried all the steps that require an user input and they are not working in the new api enpoint:

- `srs` the upload does not complete and the resume_url is not provided. we expect a redirect to` /upload/srs?id={id}`

expected:
![image](https://user-images.githubusercontent.com/19175505/150101546-502b314d-c69f-4885-9edf-430cd0cc52cd.png)

- `csv` the upload completes without asking the user to select longitude or latitude field resulting in a empty layer `/upload/csv?id={id}`

expected:
![image](https://user-images.githubusercontent.com/19175505/150101708-4e8af6c5-5acc-40c5-98b9-bdf70fe73c34.png)

- `time` the redirect_url is correct but the page used to select the date field is not rendered `/upload/time?id={id}`

expected:
![image](https://user-images.githubusercontent.com/19175505/150102628-2af6d412-9677-4611-ad78-0e208d8c5538.png)

here the file used to test these workflows:

- time -> boxes_with_date
- srs -> no-srs
- csv -> ne_110m_populated_places_simple
[dataset_with_steps.zip](https://github.com/geosolutions-it/nexus-geonode/files/7895816/dataset_with_steps.zip)

**Comment by allyoucanmap** [comment]
@giohappy @italogsfernandes I just merged the WIP of the new upload page.
Once logged in you can access the page at this path

`/catalogue/#/upload/dataset`

**Comment by italogsfernandes** [comment]
@giohappy @allyoucanmap  I just created a PR https://github.com/GeoNode/geonode/pull/8648 fixing the upload API.

I tested with the files provided by @allyoucanmap and some other files. https://github.com/geosolutions-it/nexus-geonode/issues/550#issuecomment-1016250010
The `force_ajax` param was not being respect, so this was causing your `json` page instead of a `template` when following the `redirect_to` url.

The page (`/catalogue/#/upload/dataset`) is working with the modifications in the API, but I saw 2 small issues:
* If we close the page or stop the long request, the push of the steps will be stopped.
* When uploading multiple datasets, they all start together at the same time.
